geps/gep-2643/index.md

```diff
@@ -0,0 +1,514 @@
+  // SNI attribute of TLS ClientHello message in TLS handshake. This matches
+  // the RFC 1123 definition of a hostname with 2 notable exceptions:
+  //
+  // 1. IPs are not allowed in SNI hostnames per RFC 6066.
+  // 2. A hostname may be prefixed with a wildcard label (`*.`). The wildcard
+  //    label must appear by itself as the first label.
+  //
+  // <gateway:util:excludeFromCRD>
+  // If a hostname is specified by both the Listener and TLSRoute, there
+  // must be at least one intersecting hostname for the TLSRoute to be
+  // attached to the Listener. For example:
+  //
+  // * A Listener with `test.example.com` as the hostname matches TLSRoutes
+  //   that have specified at least one of `test.example.com` or
+  //   `*.example.com`.
+  // * A Listener with `*.example.com` as the hostname matches TLSRoutes
+  //   that have specified at least one hostname that matches the Listener
+  //   hostname. For example, `test.example.com` and `*.example.com` would both
+  //   match. On the other hand, `example.com` and `test.example.net` would not
+  //   match.
+  // * A listener with `something.example.com` as the hostname matches a 
+  //   TLSRoute with hostname `*.example.com`.
+  //
+  // If both the Listener and TLSRoute have specified hostnames, any
+  // TLSRoute hostnames that do not match any Listener hostname MUST be
+  // ignored. For example, if a Listener specified `*.example.com`, and the
+  // TLSRoute specified `test.example.com` and `test.example.net`,
+  // `test.example.net` must not be considered for a match.
+  //
+  // If both the Listener and TLSRoute have specified hostnames, and none
+  // match with the criteria above, then the TLSRoute is not accepted for that 
+  // Listener. If the TLSRoute does not match any Listener on the parent, the 
+  // implementation must raise an 'Accepted' Condition with a status of
+  // `False` in the corresponding RouteParentStatus.
+  // 
+  // A Listener MUST be of type TLS when a TLSRoute attaches to it. The 
+  // implementation MUST raise an 'Accepted' Condition with a status of
+  // `False` in the corresponding RouteParentStatus with the reason 
+  // of "UnsupportedValue" in case a Listener of the wrong type is used.
+  // Core: A TLS listener CAN have mode Passthrough
+  // Extended: A TLS listener CAN have mode Terminate 
+  // </gateway:util:excludeFromCRD>
+  // +required
+  // +kubebuilder:validation:MinItems=1
+  // +kubebuilder:validation:MaxItems=16
+  Hostnames []Hostname `json:"hostnames,omitempty"`
+  // Rules is a list of TLS matchers and actions.
+  //
+  // +required
+  // +kubebuilder:validation:MinItems=1
+  // +kubebuilder:validation:MaxItems=1
+  Rules []TLSRouteRule `json:"rules,omitempty"`
+}
+
+type TLSRouteRule struct {
+  // Name is the name of the route rule. This name MUST be unique within a Route if it is set.
+  //
+  // +optional
+  Name *SectionName `json:"name,omitempty"`
+  // BackendRefs (same as other BackendRef here)
```

Comment:
If you want to use this data for load balancing, why not also specify a LB algorithm, like round-robin, random, or other?  You might also mention that the BackendRef contains a weight, if that is the only algorithmic option for traffic distribution.